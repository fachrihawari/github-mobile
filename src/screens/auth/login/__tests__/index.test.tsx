import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import LoginScreen from '../index';
import NavigationContext from '../../../../hooks/useNavigation';


describe('rendering', () => {

  let wrapper: ShallowWrapper

  function newInstance() {
    wrapper = shallow(<LoginScreen />)
  }

  beforeEach(() => {
    newInstance()
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should change username value', () => {
    const newValue = 'My new value'
    wrapper.find({ testID: 'input-username' }).simulate('changeText', newValue);

    const inputValue = wrapper.find({ testID: 'input-username' }).prop('value')
    expect(inputValue).toBe(newValue)
  })

  it('should not continue to next page', () => {
    wrapper.find({ testID: 'button-continue' }).simulate('press');
  })
})
