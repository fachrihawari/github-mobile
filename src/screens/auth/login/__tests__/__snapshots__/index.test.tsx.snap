// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoFocus={true}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Username"
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Object {
        "backgroundColor": "#f8f8f8",
        "borderColor": "#ededed",
        "borderRadius": 22.5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "fontSize": 18,
        "height": 45,
        "marginBottom": 15,
        "shadowColor": "rgba(191, 191, 191, 0.5)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 4,
        "textAlign": "center",
        "width": 600,
      }
    }
    testID="input-username"
    underlineColorAndroid="transparent"
    value=""
  />
  <Button
    onPress={[Function]}
    testID="button-continue"
    wrapperStyle={
      Object {
        "borderRadius": 22.5,
        "height": 45,
        "width": 600,
      }
    }
  >
    Continue
  </Button>
</View>
`;
